app/models/agents/liquid_output_agent.rb

```diff
@@ -129,8 +129,8 @@ def validate_options
       end
 
       if options['event_limit'].present?
-        if((Integer(options['event_limit']) rescue false) == false)
-          errors.add(:base, "Event limit must be an integer that is less than 1001.")
+	if (Integer(options['event_limit']) rescue false) == false && date_limit.blank?
```

Comment:
Can you fix the indentation here? It looks like the two lines shifted to the left.